### PR TITLE
Add support for rel/beta branch (and beta network)

### PR DESCRIPTION
**Pull request by Karmastic**
## Summary

Adds support for standing up a long-running network to do prolonged testing against a stable branch prior to releasing it to stable.

## Test Plan

Verified branch builds as desired in travis and publishes `beta` packages to S3.  All builds passed.

I've separately created an algonet-based network mirroring the configuration of devnet (5 hosts across 5 AWS regions with 1 relay and 4 nodes on each host - 20 nodes with 5% stake each).

Once this PR is merged to master, I'll delete the rel/beta branch and we'll create a new one once we're ready to snap for the next stable release.

### CODE REVIEW

File: scripts/build/nightly.sh [hunk 1]
```diff
@@ -39,7 +38,7 @@ git add ./genesistimestamp.dat ./buildnumber.dat
 git commit -m "Build ${BUILD_NUMBER} Data"
 git push
 
-TAG=rel/nightly-$(scripts/compute_build_number.sh -f)
+TAG=rel/beta-$(scripts/compute_build_number.sh -f)
```
Comment (winder) [inline scripts/build/nightly.sh]: Accidentally changed the tag here.

File: cmd/algons/dnsCmd.go [hunk 2]
```diff
@@ -268,7 +268,7 @@ func checkSrvRecord(dnsBootstrap string) {
 
 func doDeleteDNS(network string, noPrompt bool, excludePattern string) bool {
 
-	if network == "" || network == "testnet" || network == "devnet" || network == "mainnet" {
+	if network == "" || network == "testnet" || network == "devnet" || network == "mainnet" || network == "beta" {
```
Comment (tsachiherman) [inline cmd/algons/dnsCmd.go]: nit: refactor into a const map lookup.

File: scripts/build/beta.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+# Run `make` to ensure `buildtools` is available
+make -j4
+
+# Flag that we want release handling of genesis files
+export RELEASE_GENESIS_PROCESS=true
+
+git checkout rel/beta
+
+# Disabled because we have static genesis files now
```
Comment (algobolson) [inline scripts/build/beta.sh]: for this commented out block, cleanup rather than replicate?

Comment (Karmastic) [reply scripts/build/beta.sh]: Sorry - just saw this.  If i have to update again I will clean it up

File: cmd/algons/dnsCmd.go [hunk 4]
```diff
@@ -268,7 +268,8 @@ func checkSrvRecord(dnsBootstrap string) {
 
 func doDeleteDNS(network string, noPrompt bool, excludePattern string) bool {
 
-	if network == "" || network == "testnet" || network == "devnet" || network == "mainnet" {
+	validNetworks := map[string]bool{"mainnet": true, "testnet": true, "devnet": true, "betanet": true}
```
Comment (algobolson) [inline cmd/algons/dnsCmd.go]: if we're not going to make these common (here and cmd/goal/node.go ) can we at least put a comment to keep the other one up-to-date with this one (in both places) ?

**Pull request merged**